patch.c

```diff
@@ -1,6 +1,6 @@
+#include <stdlib.h>
 #include <stdio.h>
 #include <stdint.h>
-#include <alloca.h>
```

Comment:
I think maybe just wrap this in `#ifdef __Linux__`

Reply:
Although it doesn't seem to be standardised, `alloca` is not specific to Linux, but rather a GNU extension to the C compiler, implemented at least by GCC and clang, possibly other compilers. A comment in FreeBSD `stdlib.h` sheds some light:

```
/*
 * The alloca() function can't be implemented in C, and on some
 * platforms it can't be implemented at all as a callable function.
 * The GNU C compiler provides a built-in alloca() which we can use.
 * On platforms where alloca() is not in libc, programs which use it
 * will fail to link when compiled with non-GNU compilers.
 */
#if __GNUC__ >= 2
#undef  alloca  /* some GNU bits try to get cute and define this on their own */
#define alloca(sz) __builtin_alloca(sz)
#endif
```

Similar to the discussion about `sig_t`, if we define `_DEFAULT_SOURCE` then glibc `stdlib.h` will know it should pull in `alloca.h`, so that's another place where using `_DEFAULT_SOURCE` can help wuith portability.